cli/main.rs

```diff
@@ -312,6 +312,51 @@ async fn install_command(
     .map_err(ErrBox::from)
 }
 
+async fn lint_command(flags: Flags, files: Vec<String>) -> Result<(), ErrBox> {
+  let global_state = GlobalState::new(flags)?;
+
+  // TODO(bartlomieju): refactor, it's non-sense to create
+  // state just to perform unstable check...
+  use crate::state::State;
+  let state = State::new(
+    global_state.clone(),
+    None,
+    ModuleSpecifier::resolve_url("file:///dummy.ts").unwrap(),
+    None,
+    true,
+  )?;
+
+  state.check_unstable("lint");
+
+  let mut error_counts = 0;
+
+  for file in files {
+    let specifier = ModuleSpecifier::resolve_url_or_path(&file)?;
+    let source_file = global_state
+      .file_fetcher
+      .fetch_source_file(&specifier, None, Permissions::allow_all())
+      .await?;
+    let source_code = String::from_utf8(source_file.source_code)?;
+
+    let mut linter = deno_lint::linter::Linter::default();
+    let lint_rules = deno_lint::rules::get_all_rules();
+
+    let file_diagnostics = linter.lint(file, source_code, lint_rules)?;
+
+    error_counts += file_diagnostics.len();
+    for d in file_diagnostics.iter() {
+      eprintln!("{}", d.to_pretty_string());
+    }
```

Comment:
Regarding diagnostic formatting, ref https://github.com/denoland/deno_lint/pull/62.

In #4832 I made syntax errors and TS type errors look like JS runtime `error.stack` (plus the source line under the message).

<kbd><img src="https://user-images.githubusercontent.com/29990554/83910794-138f9800-a763-11ea-9cee-fd5d81409ab5.png" width=420></kbd>

<kbd><img src="https://user-images.githubusercontent.com/29990554/83910806-18ece280-a763-11ea-9744-2f6765ff311c.png" width=420></kbd>

The result has been nice. The usage range of this format covers every kind of user error and extends naturally to `deno lint` IMO.

I think it's fine if the `deno_lint` self-contained binary looks like it currently does, but this call to `d.to_pretty_string()` should be replaced by a call to the formatting logic in `cli/diagnostics.rs`.

Reply:
@nayeemrmn thanks for pointing this out. Definitely makes sense and `deno_lint` diagnostics were actually a bit problematic as they introduce potential for dependency duplication. I'll rework the PR to use Deno's diagnostic. BTW I don't think there we'll be self-contained linter binary. It looks more like a library to build your own linter ;)